core/src/main/java/org/elasticsearch/transport/TcpTransportChannel.java

```diff
@@ -23,7 +23,7 @@
 import java.io.IOException;
 import java.util.concurrent.atomic.AtomicBoolean;
 
-public final class TcpTransportChannel<Channel> implements TransportChannel {
+public final class TcpTransportChannel<Channel extends TcpChannel<Channel>> implements TransportChannel {
```

Comment:
at some point we need to get rid of the generics here. I wonder why we still need it but this is a followup...